patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -635,13 +635,25 @@
 +        }
 +        return builder.build();
 +    }
-+
++    /**
++     * Called from EntityItemFram.getAnalogOutput, used to allow items in item frames to provide their own
++     * Comparator output signal strengths, e.g. displaying any sort of world information, such as biome
++     * temperature, rainfall, etc.
++     * @param worldIn
++     * @param pos
++     * @param stack
++     * @param rotation - the default value from the ItemFrame (rotation amount)
++     * @return redstone signal strength
++     */
++    public int getItemFrameOutput(World worldIn, BlockPos pos, ItemStack stack, int rotation) {
```

Comment:
Maybe that's better. Maybe it's not. I literally have no idea.
Every method in Item already takes World/Pos parameters when appropriate, I followed that convention.